homeassistant/components/netatmo/climate.py

```diff
@@ -82,90 +148,275 @@ def temperature_unit(self):
     @property
     def current_temperature(self):
         """Return the current temperature."""
-        return self._data.current_temperature
+        return self._data.room_status[self._room_id]['current_temperature']
 
     @property
     def target_temperature(self):
         """Return the temperature we try to reach."""
-        return self._target_temperature
+        return self._data.room_status[self._room_id]['target_temperature']
 
     @property
     def current_operation(self):
         """Return the current state of the thermostat."""
-        state = self._data.thermostatdata.relay_cmd
-        if state == 0:
+        state = self._data.room_status[self._room_id]['heating_status']
+        if state is False:
             return STATE_IDLE
-        if state == 100:
+        if state is True:
             return STATE_HEAT
+        return STATE_UNKNOWN
+
+    @property
+    def operation_list(self):
+        """Return the operation modes list."""
+        return self._operation_list
+
+    @property
+    def operation_mode(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._operation_mode
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        if self._data.room_status[self._room_id]['module_type'] == \
+           'NATherm1':
+            return {
+                "home_id": self._data.homedata.gethomeId(self._data.home),
```

Comment:
It looks like a bunch of these are shared between the if/else. If possible, I'd recommend moving the shared ones to their own dict before the if and then updating that dict with the ones that are different in the if/else.

Reply:
Modified accordingly, and also pulled `module_type` to simply the code.